test/CMakeLists.txt

```diff
@@ -0,0 +1,213 @@
+     test-pipe-getsockname.c
+     test-pipe-pending-instances.c
+     test-pipe-sendmsg.c
+     test-pipe-server-close.c
+     test-pipe-set-fchmod.c
+     test-pipe-set-non-blocking.c
+     test-platform-output.c
+     test-poll-close-doesnt-corrupt-stack.c
+     test-poll-close.c
+     test-poll-closesocket.c
+     test-poll-oob.c
+     test-poll.c
+     test-process-priority.c
+     test-process-title-threadsafe.c
+     test-process-title.c
+     test-queue-foreach-delete.c
+     test-random.c
+     test-ref.c
+     test-run-nowait.c
+     test-run-once.c
+     test-semaphore.c
+     test-shutdown-close.c
+     test-shutdown-eof.c
+     test-shutdown-twice.c
+     test-signal-multiple-loops.c
+     test-signal-pending-on-close.c
+     test-signal.c
+     test-socket-buffer-size.c
+     test-spawn.c
+     test-stdio-over-pipes.c
+     test-strscpy.c
+     test-tcp-alloc-cb-fail.c
+     test-tcp-bind-error.c
+     test-tcp-bind6-error.c
+     test-tcp-close-accept.c
+     test-tcp-close-while-connecting.c
+     test-tcp-close.c
+     test-tcp-close-reset.c
+     test-tcp-connect-error-after-write.c
+     test-tcp-connect-error.c
+     test-tcp-connect-timeout.c
+     test-tcp-connect6-error.c
+     test-tcp-create-socket-early.c
+     test-tcp-flags.c
+     test-tcp-oob.c
+     test-tcp-open.c
+     test-tcp-read-stop.c
+     test-tcp-shutdown-after-write.c
+     test-tcp-try-write.c
+     test-tcp-try-write-error.c
+     test-tcp-unexpected-read.c
+     test-tcp-write-after-connect.c
+     test-tcp-write-fail.c
+     test-tcp-write-queue-order.c
+     test-tcp-write-to-half-open-connection.c
+     test-tcp-writealot.c
+     test-thread-equal.c
+     test-thread.c
+     test-threadpool-cancel.c
+     test-threadpool.c
+     test-timer-again.c
+     test-timer-from-check.c
+     test-timer.c
+     test-tmpdir.c
+     test-tty-duplicate-key.c
+     test-tty-escape-sequence-processing.c
+     test-tty.c
+     test-udp-alloc-cb-fail.c
+     test-udp-bind.c
+     test-udp-connect.c
+     test-udp-create-socket-early.c
+     test-udp-dgram-too-big.c
+     test-udp-ipv6.c
+     test-udp-multicast-interface.c
+     test-udp-multicast-interface6.c
+     test-udp-multicast-join.c
+     test-udp-multicast-join6.c
+     test-udp-multicast-ttl.c
+     test-udp-open.c
+     test-udp-options.c
+     test-udp-send-and-recv.c
+     test-udp-send-hang-loop.c
+     test-udp-send-immediate.c
+     test-udp-sendmmsg-error.c
+     test-udp-send-unreachable.c
+     test-udp-try-send.c
+     test-uname.c
+     test-walk-handles.c
+     test-watcher-cross-stop.c)
+
+add_executable(uv_run_tests ${uv_test_sources} uv_win_longpath.manifest)
+target_compile_definitions(uv_run_tests
+                           PRIVATE ${uv_defines} USING_UV_SHARED=1)
+target_compile_options(uv_run_tests PRIVATE ${uv_cflags})
+target_link_libraries(uv_run_tests uv ${uv_test_libraries})
+
+add_test(NAME uv_test
+         COMMAND uv_run_tests
+         WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})
```

Comment:
I think CI is breaking because of the change on this line?
```suggestion
         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
```

I think it is necessary not to change this because the cmake configuration doesn't currently support building out-of-tree, and probably needs something equivalent to https://github.com/libuv/libuv/blob/12be29f185261b8a7e6ada596fd805919cb2b133/configure.ac#L81-L83

Reply:
If `uv_run_tests` is run from `./test/`, the tests will fail. `PROJECT_SOURCE_DIR` sets it to the root of the project, which is what it was set to before.

Reply:
Perhaps I read the log backwards and this change is also needed on the uv_test_a target then?

Reply:
oh I see what you're talking about now. I hadn't clicked through to the jenkins CI. I'll look into this a bit more when I update the PR to resolve the current set of conflicts. The logs are either missing or I just don't have access with respect to the failures (I'm getting a 404) so I can't actually determine what caused some of them to fail.

Reply:
Direct links https://ci.nodejs.org/job/libuv-test-commit-windows-cmake/171/nodes=win2016-vs2017/console and https://ci.nodejs.org/job/libuv-test-commit-osx-cmake/146/nodes=osx1015/console
Looks like on Windows, it's an issue with the test driver script (note the different paths in the build and run steps):
```
05:51:03   uv_run_tests_a.vcxproj -> C:\workspace\libuv-test-commit-windows-cmake\out\test\Debug\uv_run_tests_a.exe
05:51:03   Building Custom Rule C:/workspace/libuv-test-commit-windows-cmake/CMakeLists.txt
05:51:03 
05:51:03 C:\workspace\libuv-test-commit-windows-cmake\out>if errorlevel 1 exit /b 
05:51:03 
05:51:03 C:\workspace\libuv-test-commit-windows-cmake\out>cd .. 
05:51:03 
05:51:03 C:\workspace\libuv-test-commit-windows-cmake>echo TAP version 13  1>out\test.tap 
05:51:03 
05:51:03 C:\workspace\libuv-test-commit-windows-cmake>out\Debug\uv_run_tests.exe   2>&1  | tee -a out\test.tap 
05:51:03 Build step 'Execute Windows batch command' marked build as failure
05:51:04 Run condition [Always] enabling perform for step [[Execute Windows batch command]]
05:51:04 [libuv-test-commit-windows-cmake] $ cmd /c call C:\Users\htqygkklxbep\AppData\Local\Temp\1\jenkins7804187108524154727.bat
```
And elsewhere, it's indeed uv_test_a, not uv_test that's failing, so probably needs your change here copied below.

Reply:
I'm receiving a 404 when I click on those links. 😓

![image](https://user-images.githubusercontent.com/63051/88741868-8f6be800-d0f5-11ea-91f5-cd909dc2a96a.png)


But regarding the Windows issue, this had happened before in a separate PR, and I believe @bnoordhuis implied it would require a change to the Jenkins configuration itself. Since I can't put that into a PR it's something someone with access would have to do. :/

Reply:
I've adjusted the permissions for https://ci.nodejs.org/job/libuv-test-commit-windows-cmake/ and https://ci.nodejs.org/job/libuv-test-commit-osx-cmake/ to be more in line with what we have for the Node.js jobs, which is that we allow authenticated users (via GitHub) to view the jobs and results. So @slurps-mad-rips you should be able to view them now.

I'll have a look at the other libuv jobs in the morning.